### DISCUSSION TITLE
Error while searching for Wii U Pro Controller

### DISCUSSION THREAD

**Discussion by JMoreno97** [KeyPuncher/WiinUPro#151] [Help]
Hi, when I try to pair my device I found this error. I have bluetooth by an USB as my computer doesn't have bluetooth itself and it is working well  but it seems like the program doesn't recognised it

![image](https://user-images.githubusercontent.com/18291802/95489754-933ea500-0997-11eb-8a8e-671e075395fb.png)

**Comment by KeyPuncher** [comment]
Hmm... There's something I could try but it's purely speculation.

But before that, can you try running it as an administrator? Perhaps it has something to do with permissions.

**Comment by JMoreno97** [comment]
I did it, I am running it as administrator.

**Comment by KeyPuncher** [comment]
Just to clarify, you did it as in running as an administrator solved your issue or you _did_ and it doesn't solve the issue?

**Comment by JMoreno97** [comment]
I meant I always run it as administrator amd it doesn't solve the problem

**Comment by LemmusLemmus** [comment]
Just for clarification, when you say that you always run it as an administrator, do you mean that you're always logged in as an administrator or that you always right-click the program/shortcut, click "Run as administrator" and accept the UAC prompt?
![image](https://user-images.githubusercontent.com/63229554/95562202-23263280-0a1c-11eb-9779-fbbb911dbe7c.png)

**Comment by JMoreno97** [comment]
I clicked on properties at the shortcut and marked an option to always run it as administrator. I can post a screenshot of the option when I arrive to the PC.

**Comment by LemmusLemmus** [comment]
Ah, ok.

**Comment by JMoreno97** [comment]
This is what I marked to always run it as administrator. I am Spanish but it says: "Run this program as administrator"
![tempsnip](https://user-images.githubusercontent.com/18291802/95590238-19b1c000-0a46-11eb-8cf4-eb045d7ad56b.png)

**Comment by KeyPuncher** [comment]
Well I though I had something but I looked at the code again and I don't have any coding ideas right now.

Try going to Windows Bluetooth and then let it search for devices like you are going to add a device, then while it is searching, use the sync feature in WiinUSoft.

**Comment by JMoreno97** [comment]
I tried it and still not working.
Also if I try to connect via Windows Bluetooth it asks me for a code and I don't know what code could be (I tried 0000 and 1234)
![Captura](https://user-images.githubusercontent.com/18291802/95738455-8cf74400-0c89-11eb-97e6-ab9ba38e125c.PNG)

**Comment by LemmusLemmus** [comment]
IIRC the default code is nothing (no text whatsoever). Be aware that the settings app doesn't let you enter nothing, so you will have to use the control panel, as you did in the picture above.

**Comment by JMoreno97** [comment]
I dont see how can introduce nothing 😅

**Comment by LemmusLemmus** [comment]
When it asks for a code simply do not enter one. Only press the continue/next button.

**Comment by JMoreno97** [comment]
It doesn't let me click continue if I let it blank

**Comment by LemmusLemmus** [comment]
Have you tried connecting the controller using the old control panel instead? The new windows settings app doesn't let you enter blank.

**Comment by JMoreno97** [comment]
I have tried from 2 sites.
The first one is what I think you mean as old control panel. It doesn't let me leave enter blank.
The second one is the other W10 panel and it seems that bluetooth is not been detected here.
![Captura](https://user-images.githubusercontent.com/18291802/95748995-2bd86c00-0c9b-11eb-9464-412e58589d0b.PNG)
![captura2](https://user-images.githubusercontent.com/18291802/95748852-f59aec80-0c9a-11eb-9804-190341bccf56.PNG)

**Comment by LemmusLemmus** [comment]
Unlike your control panel I don't have the Bluetooth Devices option (in red):
![image](https://user-images.githubusercontent.com/63229554/95763529-6e0ba880-0caf-11eb-81f8-3ef0797e6259.png)
I added my controller by clicking the "Agregar un dispositivo" option (in green). Try doing that instead, this window should hopefully appear:
![image](https://user-images.githubusercontent.com/63229554/95763333-300e8480-0caf-11eb-90e0-0db66e7e7d27.png)
Upon selecting the controller this window opens for me: 
![image](https://user-images.githubusercontent.com/63229554/95762658-4e27b500-0cae-11eb-9ce3-935e652cca27.png)
Unlike elsewhere I can actually click the "Next" button here without providing a code. See if that works.

Bluetooth not working in the settings app would seem to indicate that the problem lies with Windows and not specifically with WiinUPro. If the process I described above doesn't work it might be worth attempting to update the Bluetooth drivers through the device manager, but I honestly don't know if that would actually solve the problem since I've got no idea as to what may be the underlying problem here.
![image](https://user-images.githubusercontent.com/63229554/95764314-95af4080-0cb0-11eb-87ba-dde3c159c49b.png)

**Comment by JMoreno97** [comment]
I tried to search the device as you said but it doesn't recognise it.
Also I tried to update my bluetooth driver and it is updated. My driver name isn't as yours, I don't know if it could be because my bluetooth is been given by a pen drive. Maybe W10 doesn't recognised it well.

![Captura](https://user-images.githubusercontent.com/18291802/95857122-70c3d780-0d5b-11eb-9950-6cbdbf705611.PNG)
![Captura](https://user-images.githubusercontent.com/18291802/95857139-76212200-0d5b-11eb-9335-660df00a0962.PNG)

**Comment by LemmusLemmus** [comment]
I also have an external Bluetooth dongle, so that shouldn't be the problem here. You can attempt to change the driver used by your Bluetooth Radio, from "CSR Bluetooth Radio" to "Generic Bluetooth Radio". Windows 10 will hopefully recognize your Bluetooth device then.
If you wish to try this: go to the device manger, right-click the "CSR Bluetooth Radio" item, press the update button and select "Browse my computer for drivers".
![image](https://user-images.githubusercontent.com/63229554/95893750-61a74e80-0d88-11eb-9ebf-f63d4a4d57a2.png)
Click the option "Let me pick from a list of available drivers on my computer".
![image](https://user-images.githubusercontent.com/63229554/95893923-a9c67100-0d88-11eb-8250-e28ec53034ec.png)
**If** it is in the list of compatible drivers, choose the "Generic Bluetooth Radio" option (or maybe it's the "Generic Bluetooth Adapter" in your case, you _can_ try that option if "Generic Bluetooth Radio" doesn't work). If it's not in the list I'd imagine this potential solution won't work.
![image](https://user-images.githubusercontent.com/63229554/95894044-d5495b80-0d88-11eb-8a40-e361a8f63843.png)

When you are done you might have to restart your computer for the changes to take effect. Windows 10 should hopefully detect your Bluetooth device now, but I can't promise anything.

[Link to where I found this potential solution.](https://answers.microsoft.com/en-us/windows/forum/windows_10-hardware-winpc/csr-v40-bluetooth-dongle-does-not-work-on-my/fefc9f9a-22b5-4b75-9da4-c1c880794a99)

**Comment by JMoreno97** [comment]
I found that solution too but I don't have the Generic bluetooth Radio driver
![Captura](https://user-images.githubusercontent.com/18291802/96265184-6e64b600-0fc5-11eb-8880-7328e56cfdc1.PNG)

**Comment by LemmusLemmus** [comment]
Hmm.
[See if any of these methods work first, unless you've already tried them all.](https://answers.microsoft.com/en-us/windows/forum/windows_7-hardware/generic-bluetooth-adapter-not-found-in-device/4b322c73-1e60-4784-b98e-9b580bd36ebb)

I'd assume we've identified the problem here, but if these potential solutions don't work I'm pretty much all out of ideas on how to solve it. The driver itself might not even be the issue here, I can't know for sure. You could attempt to uncheck the "Mostrar el hardware compatible" checkbox and then choose the "Generic Bluetooth Driver" from the new list which appears, but using that driver might cause issues.

**Comment by JMoreno97** [comment]
I still can't connect :s

**Comment by gigglebunsuv** [comment]
I know I am super late, but to install the generic bluetooth radio, you have to go to Universal Serial Bus in device manager and find the CSR USB device, and you can change it there.
